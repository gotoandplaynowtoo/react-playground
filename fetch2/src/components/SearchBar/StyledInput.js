import Input from 'components/Input';

const StyledInput = Input.extend`
  margin-right: 6px;
`;

export default StyledInput;
