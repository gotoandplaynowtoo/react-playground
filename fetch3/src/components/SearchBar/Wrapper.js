import styled from 'styled-components';

const Wrapper = styled.div`
  padding-top: 16px;
  display: flex;
  justify-content: center;
  align-items: center;
`;

export default Wrapper;