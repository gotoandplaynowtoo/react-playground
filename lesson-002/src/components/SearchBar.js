import React, { Component } from 'react';

class Test extends Component {
  constructor(props) {
    super(props);
    console.log('instantiated');
  }

  render = () => {
    console.log('i am rerendered too');
    return <p>Lorem ipsum dolor sit, amet consectetur adipisicing elit. Cum enim illo excepturi exercitationem rem eligendi dolores, qui ab porro facere quae error ea adipisci vitae blanditiis ipsum sit voluptatibus. Aliquam.</p>
  };
}

class SearchBar extends Component {

  constructor(props) {
    super(props);
    this.state = {
      term: ''
    };
  }

  onInputChange = (e) => {
    this.setState({
      term: e.target.value
    });
  };

  render = () => {
    return (
      <div>
        <input 
          value={this.state.term}
          onChange={this.onInputChange} 
        />
        <Test/>
      </div>
    );
  }
}

export default SearchBar;