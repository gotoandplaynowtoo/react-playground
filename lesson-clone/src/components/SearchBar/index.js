import React from 'react';
import Input from 'components/Input';
import Button from 'components/Button';

const SearchBar = (props) => {
  return (
    <form>
      <Input placeholder='Search...'/>
      <Button>Search</Button>
    </form>
  );
};

export default SearchBar;