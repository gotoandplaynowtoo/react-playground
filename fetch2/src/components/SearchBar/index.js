import React from 'react';
import Button from 'components/Button';
import StyledInput from './StyledInput';
import Wrapper from './Wrapper';

const SearchBar = (props) => {
  const {onSubmit, onChange, value} = props;
  return (
    <Wrapper>
      <form onSubmit={onSubmit}>
        <StyledInput onChange={onChange} value={value}/>
        <Button type='submit'>Search</Button>
      </form>
    </Wrapper>
  );
}



export default SearchBar;