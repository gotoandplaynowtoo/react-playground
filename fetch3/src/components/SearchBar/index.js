import React from 'react';
import Input from 'components/Input';
import Button from 'components/Button';
import Wrapper from './Wrapper';

const SearchBar = (props) => {

  const { 
    onSubmit,
    onChange, 
    defaultValue
  } = props

  return (
    <Wrapper>
      <form 
        onSubmit={onSubmit}
        onChange={onChange}
      >
        <Input name='searchTerm' defaultValue={defaultValue}/>
        <Button type='submit'>Search</Button>
      </form>
    </Wrapper>
  );
};

export default SearchBar;