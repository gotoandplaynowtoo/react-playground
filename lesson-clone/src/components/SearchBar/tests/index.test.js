import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import SearchBar from 'components/SearchBar';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<SearchBar/>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('render correctly', () => {
  const tree = renderer.create(<SearchBar/>).toJSON();
  expect(tree).toMatchSnapshot();
});