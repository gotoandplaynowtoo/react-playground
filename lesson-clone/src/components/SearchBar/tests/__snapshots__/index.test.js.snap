// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<form>
  <input
    className="sc-bdVaJa kSylSs"
    placeholder="Search..."
  />
  <button
    className="sc-bwzfXH iqmKiI"
  >
    Search
  </button>
</form>
`;
